packages/discovery-provider/ddl/migrations/0102_shorten_challenge_ids.sql

```diff
@@ -0,0 +1,246 @@
+/** 
+ * Migrates our existing challenge IDs to shorter ones, and moves specifiers to 
+ * hex values so that they are shorter (when stringified) too.
+ * This allows us to put challenge claims into single Solana transactions as
+ * the disbursement IDs need to be less than 21 bytes.
+ * 
+ * This migration turns off triggers, drops indexes, and drops pkey constraints
+ * to increase speed, and carefully readds them in the end.
+**/
+
+BEGIN;
+
+/* CHALLENGES */
+
+-- remove foreign key
+ALTER TABLE user_challenges DROP CONSTRAINT user_challenges_challenge_id_fkey;
+
+-- Update challenges table
+UPDATE challenges SET id = 'p' WHERE id = 'profile-completion';
+UPDATE challenges SET id = 'l' WHERE id = 'listen-streak';
+UPDATE challenges SET id = 'u' WHERE id = 'track-upload';
+UPDATE challenges SET id = 'r' WHERE id = 'referrals';
+UPDATE challenges SET id = 'rv' WHERE id = 'ref-v';
+UPDATE challenges SET id = 'rd' WHERE id = 'referred';
+UPDATE challenges SET id = 'v' WHERE id = 'connect-verified';
+UPDATE challenges SET id = 'm' WHERE id = 'mobile-install';
+UPDATE challenges SET id = 'ft' WHERE id = 'send-first-tip';
+UPDATE challenges SET id = 'fp' WHERE id = 'first-playlist';
+
+/* USER CHALLENGES */
+
+-- disable trigger (doesn't exist on test db?)
+DO $$
+BEGIN
+    IF EXISTS (
+        SELECT 1 
+        FROM information_schema.triggers 
+        WHERE event_object_table = 'user_challenges' 
+        AND trigger_name = 'on_user_challenge'
+    ) THEN
+        EXECUTE 'ALTER TABLE user_challenges DISABLE TRIGGER on_user_challenge';
+    END IF;
+END $$;
+
+-- remove pkey from user_challenges before triggering constraint checks
+ALTER TABLE user_challenges DROP CONSTRAINT user_challenges_pkey;
+
+-- remove indexes
+DROP INDEX user_challenges_challenge_idx;
+
+-- Update user_challenges, using the challenge id as a filter to ensure idempotency
+UPDATE user_challenges 
+  SET 
+    challenge_id = 'p', 
+    specifier = to_hex(specifier::integer)
```

Comment:
noticed that you did a `[2:]` in python which i'm guessing is to get rid of the `0x` prefix, does `to_hex` omit the `0x`?

Reply:
Yep!